src/update_agent/mod.rs

```diff
@@ -79,14 +79,16 @@ pub struct InteractiveSession {
     tty_dev: String,
 }
 
-/// Function to deserialize field to `Option<String>`, where empty strings are
-/// deserialized into `None`.
-fn empty_string_as_none<'de, D>(deserializer: D) -> Result<Option<String>, D::Error>
+/// Function to deserialize field to `Option<String>`, where empty strings or
+/// `n/a` (not available) strings are deserialized into `None`. In systemd v254+
+/// loginctl list-sessions --json started outputting `n/a` instead of an empty
+/// string for tty.
+fn empty_or_na_string_as_none<'de, D>(deserializer: D) -> Result<Option<String>, D::Error>
 where
     D: Deserializer<'de>,
 {
     let s = String::deserialize(deserializer)?;
-    if s.is_empty() {
+    if s.is_empty() || s.eq(&String::from("n/a")) {
```

Comment:
```suggestion
    if s.is_empty() || s == "n/a" {
```

Reply:
(Not only is this shorter, `String::from` is doing an avoidable heap allocation)

Reply:
I've been exposed! I need to learn more rust!

Reply:
Fixed!